site-builder/src/site/resource.rs

```diff
@@ -344,15 +344,17 @@ impl ResourceManager {
             extension.to_string_lossy(),
             full_path.to_string_lossy()
         ))?)?;
-        // TODO(giac): this could be (i) async; (ii) pre configured with the number of shards to
-        //     avoid chain interaction (maybe after adding `info` to the JSON commands).
-        let output = self.walrus.blob_id(full_path.to_owned(), None)?;
-        // Need to check the size to avoid calling encode on empty files.
+
         let plain_content = std::fs::read(full_path)?;
         if plain_content.is_empty() {
-            // We are ignoring empty files.
-            return Ok(None);
+            let error_message = format!("The file {} is empty.", full_path.to_string_lossy());
+            return Err(
+                anyhow!(error_message).context("Walrus sites does not support empty files.")
```

Comment:
Suggestion: AFAIU, `context` is added when propagating errors. Since here you are creating the error, you could have the full message saying `The file {} is empty; Walrus does not support..`. Or is there another reason to have the two-level error message?

Reply:
I did this actually because it looked readable to me, but on second thought context is better to be use for propagating errors as you suggest.